core/trino-spi/src/main/java/io/trino/spi/eventlistener/TableReferenceInfo.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.eventlistener;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * This class is JSON serializable for convenience and serialization compatibility is not guaranteed across versions.
+ *
+ * @see TableInfo#getReferenceChain()
+ */
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "@type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = ViewReferenceInfo.class, name = "view"),
+        @JsonSubTypes.Type(value = RowFilterReferenceInfo.class, name = "rowFilter"),
+        @JsonSubTypes.Type(value = ColumnMaskReferenceInfo.class, name = "columnMask")})
```

Comment:
https://github.com/airlift/airlift/commit/d4e8ad91004a05c19015146bcffede5c3f9fa787#diff-d272f3f6c0303cb179e0582fded0e5ed882dfdf944eef173dfcb6725d4a3e1d8

This could potentially allow us to remove these annotation

Reply:
No changes requested for my reference

Reply:
cool!